core/store/migrate/migrations/0141_allow_broadcast_at_on_fatally_errored_tx.sql

```diff
@@ -0,0 +1,12 @@
+-- +goose Up
+ALTER TABLE eth_txes DROP CONSTRAINT chk_eth_txes_fsm;
+ALTER TABLE eth_txes ADD CONSTRAINT chk_eth_txes_fsm CHECK (
+    state = 'unstarted'::eth_txes_state AND nonce IS NULL AND error IS NULL AND broadcast_at IS NULL AND initial_broadcast_at IS NULL OR state = 'in_progress'::eth_txes_state AND nonce IS NOT NULL AND error IS NULL AND broadcast_at IS NULL AND initial_broadcast_at IS NULL OR state = 'fatal_error'::eth_txes_state AND nonce IS NULL AND error IS NOT NULL OR state = 'unconfirmed'::eth_txes_state AND nonce IS NOT NULL AND error IS NULL AND broadcast_at IS NOT NULL AND initial_broadcast_at IS NOT NULL OR state = 'confirmed'::eth_txes_state AND nonce IS NOT NULL AND error IS NULL AND broadcast_at IS NOT NULL AND initial_broadcast_at IS NOT NULL OR state = 'confirmed_missing_receipt'::eth_txes_state AND nonce IS NOT NULL AND error IS NULL AND broadcast_at IS NOT NULL AND initial_broadcast_at IS NOT NULL
+) NOT VALID; -- NOT VALID gives large speedup and this is a relaxing of the constraint so its safe
```

Comment:
nit/ could format this for readability. https://codebeautify.org/sqlformatter did this:
```suggestion
ALTER TABLE eth_txes ADD CONSTRAINT chk_eth_txes_fsm CHECK (
    state = 'unstarted' :: eth_txes_state 
    AND nonce IS NULL 
    AND error IS NULL 
    AND broadcast_at IS NULL 
    AND initial_broadcast_at IS NULL 
    OR state = 'in_progress' :: eth_txes_state 
    AND nonce IS NOT NULL 
    AND error IS NULL 
    AND broadcast_at IS NULL 
    AND initial_broadcast_at IS NULL 
    OR state = 'fatal_error' :: eth_txes_state 
    AND nonce IS NULL 
    AND error IS NOT NULL 
    OR state = 'unconfirmed' :: eth_txes_state 
    AND nonce IS NOT NULL 
    AND error IS NULL 
    AND broadcast_at IS NOT NULL 
    AND initial_broadcast_at IS NOT NULL 
    OR state = 'confirmed' :: eth_txes_state 
    AND nonce IS NOT NULL 
    AND error IS NULL 
    AND broadcast_at IS NOT NULL 
    AND initial_broadcast_at IS NOT NULL 
    OR state = 'confirmed_missing_receipt' :: eth_txes_state 
    AND nonce IS NOT NULL 
    AND error IS NULL 
    AND broadcast_at IS NOT NULL 
    AND initial_broadcast_at IS NOT NULL
) NOT VALID; -- NOT VALID gives large speedup and this is a relaxing of the constraint so its safe
```
but one line per state might be better? :shrug: